src/rfc-8252-http-server.ts

```diff
@@ -61,6 +61,10 @@ export class RFC8252HTTPServer {
       (resolve, reject) =>
         ([this.oidcParamsResolve, this.oidcParamsReject] = [resolve, reject])
     );
+    // Errors are handled by the caller of this instance.
+    this.oidcParamsPromise.catch(() => {
```

Comment:
Did oidcParamsPromise reject before something had a chance to call waitForOIDCParams? Maybe other awaits in between causing ticks. Worries me that we don't fully understand why this would happen.

Reply:
It’s almost certainly this `await` that’s delaying the call to `waitForOIDCParams`: https://github.com/mongodb-js/oidc-plugin/blob/8e78b9e1686a35c925c1d11987eeb2a384b850e4/src/plugin.ts#L325-L330

That’s a call that’s expected to take a while to resolve (because it waits for the browser to start and access the local server), so it’s not really surprising that it does happen, it’s surprising that I’m _not_ seeing a local warning outside of Windows CI.

I think this is still the semantically right thing to do either way; a rejection of `oidcParamsPromise` does not indicate a programming error, it indicates a user/security/runtime error, which shouldn’t result in an unhandled promise rejection that emits a warning or stops the process.